source/common/router/rds_impl.cc

```diff
@@ -194,11 +205,30 @@ Router::ConfigConstSharedPtr RdsRouteConfigProviderImpl::config() {
   return tls_->getTyped<ThreadLocalConfig>().config_;
 }
 
+bool RdsRouteConfigProviderImpl::requestVirtualHostsUpdate(const std::string& for_domain,
+                                                           std::function<void()> cb) {
+  if (!config()->usesVhds()) {
+    return false;
+  }
+  factory_context_.dispatcher().post(
+      [this, for_domain]() -> void { subscription_->ondemandUpdate({for_domain}); });
```

Comment:
This should be `onDemandUpdate`..

Reply:
There's an issue of collision with Envoy callback naming, though, right? `onDemandUpdate` sounds like a function that handles a "demand update". Not sure that there's a good solution here, hahaha....

Reply:
Maybe `updateOnDemand`? `lazyUpdate`?